codice/ddf
# DDF-4465 Export metacards to a compressed file in a specified format

#### What does this PR do?
- Modifies `ZipCompressor` so that it no longer requires an absolute file path to function properly. 
- Adds the ability to export the metacards with a different file format such as HTML. 

#### Who is reviewing it? 

@GabrielFabian 
@cantstoptheunk 
@derekwilhelm 

#### Ask 2 committers to review/merge the PR and tag them here.
@jlcsmith 
@rzwiefel 

#### How should this be tested?

- Ingest a couple of metacards
- Invoke the query response transformer. You can use this postman request. 
- The CQL in the attached request matches everything. 
- Note that you may have to add a `.zip` onto the downloaded file to be able to unpack it. This is not a failure of the transformer but one of the client that you're using. 
- Verify that no regressions in the `catalog:dump` command exist. 

#### Any background context you want to provide?
- We wanted the ability to export a result set as a compressed file. 

#### What are the relevant tickets?
[DDF-4465](https://codice.atlassian.net/browse/DDF-4465)

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
